src/components/AssetsList.vue

```diff
@@ -0,0 +1,642 @@
+            </td>
+            <td class="asset-input" v-if="asset.buyInput" @click.stop>
+              <SmallBlock
+                v-on:close="() => { asset.buyInput = false; assetList = [...assetList] }">
+                <CurrencyForm
+                  label="Buy"
+                  :symbol="asset.symbol"
+                  :price="asset.price"
+                  :hasSecondButton="true"
+                  v-on:submitValue="(value) => investValue(asset, value)"
+                  :waiting="waitingForInvest"
+                  flexDirection="row"
+                  :validators="[
+                    {require: value => assetList[0].balance >= asset.price * value, message: 'Requested asset value exceeds your available AVAX balance'},
+                  ]"
+                  :info="(value) => `Amount in AVAX: <b>${(asset.price * value).toFixed(2)}</b>`"
+                />
+              </SmallBlock>
+            </td>
+            <td class="chart" v-if="(asset.showChart || isMobile) && asset.prices" @click.stop>
+              <SmallBlock
+                v-on:close="() => { asset.showChart = false; assetList = [...assetList] }">
+                <Chart
+                  :dataPoints="asset.prices"
+                  :minY="asset.minPrice" :maxY="asset.maxPrice" :lineWidth="3"/>
+              </SmallBlock>
+            </td>
+          </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
+  </div>
+</template>
+
+
+<script>
+  import Chart from "@/components/Chart.vue";
+  import SimpleChart from "@/components/SimpleChart.vue";
+  import Block from "@/components/Block.vue";
+  import CurrencyForm from "@/components/CurrencyForm.vue";
+  import SmallBlock from "@/components/SmallBlock.vue";
+  import { mapState, mapActions } from "vuex";
+  import redstone from 'redstone-api';
+
+
+  export default {
+    name: 'AssetsList',
+    components: {
+      Chart,
+      Block,
+      CurrencyForm,
+      SimpleChart,
+      SmallBlock
+    },
+    props: {
+      assets: [],
+      fields: [
+        'Asset',
+        'Price',
+        'Balance',
+        'Value',
+        'Share',
+        { key: 'actions', label: ''}
+      ],
+    },
+    computed: {
+      ...mapState('loan', ['totalValue']),
+      investments() {
+        return this.assetList.filter(
+          asset => {
+            return asset.balance > 0 || asset.native
+          }
+        )
+      },
+      options() {
+        return this.assetList.filter(
+          asset => {
+            return asset.balance === 0 && !asset.native
+          }
+        )
+      }
+    },
+    data() {
+      return {
+        assetList: [],
+        minimumValue: 0,
+        maximumValue: 0,
+        waitingForInvest: false,
+        waitingForRedeem: false
+      }
+    },
+    methods: {
+      ...mapActions('loan', ['invest', 'redeem']),
+      toggleChart(asset) {
```

Comment:
Hmm, I think the approach of passing variable reference and then refreshing the list makes the code quite unreadable. Can we use an object instead of the array and identify assets by symbol?